server/tm_cmds.go

```diff
@@ -115,3 +122,130 @@ func VersionCmd() *cobra.Command {
 		},
 	}
 }
+
+// QueryBlocksByEventsCmd returns a command to search through blocks by events.
+func QueryBlocksByEventsCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "blocks",
+		Short: "Query for paginated blocks that match a set of events",
+		Long: `Search for blocks that match the exact given events where results are paginated.
+The events query is directly passed to Tendermint's RPC BlockSearch method and must
+conform to Tendermint's query syntax.
+Please refer to each module's documentation for the full set of events to query
+for. Each module documents its respective events under 'xx_events.md'.
+`,
+		Example: fmt.Sprintf(
+			"$ %s query blocks --query \"message.sender='cosmos1...' AND block.height > 7\" --page 1 --limit 30 --order-by ASC",
+			version.AppName,
+		),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientQueryContext(cmd)
+			if err != nil {
+				return err
+			}
+			query, _ := cmd.Flags().GetString(auth.FlagQuery)
+			page, _ := cmd.Flags().GetInt(flags.FlagPage)
+			limit, _ := cmd.Flags().GetInt(flags.FlagLimit)
+			orderBy, _ := cmd.Flags().GetString(auth.FlagOrderBy)
```

Comment:
Let's not bring in auth just to use a flag. We can redefine them here 👍

Reply:
I'm happy to do whatever @alexanderbez  - lmk if you change your mind based on @julienrbrt comment

Reply:
Leaving it in auth for now